src/features/Courses/AddClass/index.jsx

```diff
@@ -56,12 +53,7 @@ const AddClass = ({
       } catch (error) {
         logError(error);
       } finally {
-        if (isDetailClassPage) {
-          dispatch(fetchAllClassesData(selectedInstitution.id, courseInfo.masterCourseId));
-        } else {
-          dispatch(fetchClassesData(selectedInstitution.id, initialPage, courseInfo.masterCourseId));
-          dispatch(updateClassesCurrentPage(initialPage));
-        }
+        finalCall(selectedInstitution.id, initialPage, courseInfo.masterCourseId);
```

Comment:
It's not necessary to send the institution id and initial page, that info could be access from every page because institution id is in the redux store and initial page is a constant. Please remove it to avoid this arguments  and update all finalCall from the others components